### PR TITLE
Add documentation for Boogie command line options

### PR DESCRIPTION
**Pull request by atomb**
Several Boogie command-line options are very useful for Dafny and therefore worth mentioning in the Dafny reference manual. The options included here especially focus on stability analysis.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/UserGuide.md [hunk 1]
```diff
@@ -1385,7 +1385,65 @@ and what information it produces about the verification process.
...
+### 24.10.8. Controlling Boogie {#sec-controlling-boogie}
+
+Dafny builds on top of Boogie, a general-purpose intermediate language
+for verification. Options supported by Boogie on its own are also
+supported by Dafny. Some of the Boogie options most relevant to Dafny
+users include the following. We use the term "procedure" below to refer
+to a Dafny function, lemma, method, or predicate, following Boogie
+terminology.
+
+* `-proc:<name>` - verify only the procedure named `<name>`. The name
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:1397]: Wow I can't believe this wasn't documented before! Well done!

File: docs/DafnyRef/UserGuide.md [hunk 2]
```diff
@@ -1385,7 +1385,65 @@ and what information it produces about the verification process.
...
+  proof attempt will use a new random seed derived from this original
+  seed. If not, it will implicitly use `-randomSeed:0` to ensure a
+  difference between iterations. This option can be very useful for
+  identifying input programs for which verification is unstable. If the
+  verification times or solver resource counts associated with each
+  proof attempt vary widely for a given procedure, small changes to that
+  procedure might be more likely to cause proofs to fail in the future.
+
+* `-vcsSplitOnEveryAssert` - prove each (explicit or implicit) assertion
+  in each procedure separately. By default, Boogie attempts to prove
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:1428]:
```suggestion
  in each procedure separately. See also the attribute [`{:vcs_split_on_every_assert}`](#sec-vcs_split_on_every_assert) for restricting this option on specific procedures. By default, Boogie attempts to prove
```
```suggestion
  in each procedure separately. By default, Boogie attempts to prove
```

File: docs/DafnyRef/UserGuide.md [hunk 3]
```diff
@@ -1385,7 +1385,65 @@ and what information it produces about the verification process.
...
+  sometimes causes the solver to take longer. If a proof that you
+  believe should succeed is timing out, using this option can sometimes
+  help.
+
+* `-vcsCores:<n>` - try to verify `<n>` procedures simultaneously.
+  Setting `<n>` to the number of physical cores available tends to be
+  effective at speeding up overall proof time.
+
+* `-timeLimit:<n>` - spend at most `<n>` seconds attempting to prove any
+  single SMT query.
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:1440]:
```suggestion
  single SMT query. This setting can also be set per method using the attribute [`{:timeLimit n}`](#sec-time-limit).
```
Also, can you please ensure the link is valid? I think you'll have to insert `{#sec-time-limit}` at the right place in `Attributes.md`

File: docs/DafnyRef/UserGuide.md [hunk 4]
```diff
@@ -1385,7 +1385,65 @@ and what information it produces about the verification process.
...
+
+* `-rlimit:<n>` - set the maximum solver resource count to use while
+  proving a single SMT query. This can be a more deterministic approach
+  than setting a time limit. To choose an appropriate value, try proving
+  some representative code and comparing the time and resource count
+  values reported to calculate a correspondence.
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:1446]:
```suggestion
  than setting a time limit. To choose an appropriate value, please refer to the documentation of the attribute [`{:rlimit}`](#sec-rlimit) that can be applied per procedure.
```

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
This is a great addition ! A few updates to enhance it a bit.

File: docs/DafnyRef/UserGuide.md [hunk 5]
```diff
@@ -1385,7 +1385,63 @@ and what information it produces about the verification process.
...
+  single query to an SMT solver. This usually performs well, but
+  sometimes causes the solver to take longer. If a proof that you
+  believe should succeed is timing out, using this option can sometimes
+  help.
+
+* `-vcsCores:<n>` - try to verify `<n>` procedures simultaneously.
+  Setting `<n>` to the number of physical cores available tends to be
+  effective at speeding up overall proof time.
+
+* `-timeLimit:<n>` - spend at most `<n>` seconds attempting to prove any
```
Comment (davidcok) [inline docs/DafnyRef/UserGuide.md:1439]: The Dafny Attributes.md file mentions "Boogie `timelimit` command-line option"

Is it -timelimit or -timeLimit? Please fix the line that is wrong.

**Review by davidcok — CHANGES_REQUESTED** [review]

File: docs/DafnyRef/UserGuide.md (hunk 5 above)
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:1518]: Done.

File: docs/DafnyRef/Attributes.md [hunk 6]
```diff
@@ -419,16 +419,16 @@ There are also two different approaches to executing all tests in a program:
    This runner is currently very basic, but avoids introducing any additional target
    language dependencies in the compiled code.
 
-### 23.2.15. `{:timeLimit N}`
+### 22.2.14. `{:timeLimit N}` {#sec-time-limit}
```
Comment (davidcok) [inline docs/DafnyRef/Attributes.md:422]: This reversion of the section number can't be correct. Are you sure you are in synch with master? or did you add a section?  Run 'make numbers' in DafnyRef to recompute all section numbers.

**Review by davidcok — APPROVED** [review]

File: docs/DafnyRef/UserGuide.md [hunk 7]
```diff
@@ -1516,13 +1516,11 @@ terminology.
   effective at speeding up overall proof time.
 
 * `-timeLimit:<n>` - spend at most `<n>` seconds attempting to prove any
-  single SMT query.
+  single SMT query. This setting can also be set per method using the attribute [`{:timeLimit n}`](#sec-time-limit).
 
 * `-rlimit:<n>` - set the maximum solver resource count to use while
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:1521]: I think this sentence needs a full stop

**Review by MikaelMayer — CHANGES_REQUESTED** [review]

File: docs/DafnyRef/UserGuide.md (hunk 7 above)
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:1521]: It has one, though, doesn't it?

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**